scio-core/src/main/java/com/spotify/scio/transforms/BaseAsyncLookupDoFn.java

```diff
@@ -345,44 +357,26 @@ public boolean equals(Object obj) {
   /**
    * {@link Cache} supplier for {@link BaseAsyncLookupDoFn}.
    *
-   * @param <A> input element type.
-   * @param <B> lookup value type.
    * @param <K> key type.
+   * @param <V> value type
    */
-  public abstract static class CacheSupplier<A, B, K> implements Serializable {
-    /**
-     * Create a new {@link Cache} instance. This is called once per {@link BaseAsyncLookupDoFn}
-     * instance.
-     */
-    public abstract Cache<K, B> createCache();
-
-    /** Get cache key for the input element. */
-    public abstract K getKey(A input);
```

Comment:
:warning: This is a breaking change :warning: 
I feel this API was very complex. To support this, users would have to provide their own cache implementation that would delegate to an underlying cache.

Reply:
sounds good, since `BaseAsyncLookupDoFn` now extends `DoFnWithResource` and users don't have to worry about the `UUID` key, I think it makes sense to get rid of this complex logic!